### PR TITLE
Support modern IPython

### PR DESCRIPTION
**Pull request by matan129**
Solves #91!

I had to jump through hoops to keep everything compatible with Python 2. It might be a good idea to remove support for it all together and bump the major version of `doitlive`.

### CODE REVIEW

**Comment by matan129** [comment]
@sloria thoughts? 🙂

File: azure-pipelines.yml [hunk 1]
```diff
@@ -18,7 +18,27 @@ variables:
...
+      py37-ipython7,
+
+      py38-ipython7,
+      py38-ipython8,
+
+      py39-ipython7,
+      py39-ipython8,
+
+      py310-ipython7,
+      py310-ipython8
```
Comment (sloria) [inline azure-pipelines.yml:40]: suggestion (non-blocking): probably overkill to test every supported combination. imo it'd suffice to do

```suggestion
      py27-ipython5,
      
      py36-ipython5,
      py36-ipython6,

      py310-ipython7,
      py310-ipython8
```

Comment (sloria) [reply azure-pipelines.yml:40]: that said, we should definitely drop support for EOL'd python and versions in a follow-up PR

File: tox.ini [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+deps:
+  ipython5: IPython>=5,<6
+  ipython6: IPython>=6,<7
+  ipython7: IPython>=7,<8
+  ipython8: IPython>=8
+
 extras = tests
 commands = pytest -s {posargs}
-passenv = HOME SHELL CI
+passenv = HOME SHELL CI DOITLIVE_INTERPRETER
```
Comment (sloria) [inline tox.ini:13]: question (non-blocking): is this change necessary?

**Review by sloria — APPROVED** [review]
this looks great! sorry for the delay and thanks for the ping--i honestly just missed this is my feed.

i'll go ahead and cut a release with this.

File: tox.ini (hunk 2 above)
Comment (matan129) [reply tox.ini:13]: Yeah. In my setup, the default shell is `fish`, but the tested scripts use bash. Since fish is not bash (or POSIX) compatible, this breaks, so when testing locally,  I wanted to override `fish` with `bash`.
